pkg/api/goprobe/client/query.go

```diff
@@ -0,0 +1,46 @@
+package client
+
+import (
+	"context"
+
+	gpapi "github.com/els0r/goProbe/pkg/api/goprobe"
+	"github.com/els0r/goProbe/pkg/query"
+	"github.com/els0r/goProbe/pkg/results"
+	"github.com/fako1024/httpc"
+)
+
+// Run implements the query.Runner interface
+func (c *Client) Run(ctx context.Context, args *query.Args) (*results.Result, error) {
+	return c.Query(ctx, args)
+}
+
+// Query runs a query on the API endpoint
+func (c *Client) Query(ctx context.Context, args *query.Args) (*results.Result, error) {
+	// use a copy of the arguments, since some fields are modified by the client
+	queryArgs := *args
+	// whatever happens, the results are expected to be returned in json
+	queryArgs.Format = "json"
+
+	if queryArgs.Caller == "" {
+		queryArgs.Caller = clientName
+	}
+
+	// we need more results before truncating
+	if args.NumResults < query.DefaultNumResults {
+		args.NumResults = query.DefaultNumResults
+	}
+
+	var res = new(results.Result)
+
+	req := c.Modify(ctx,
+		httpc.NewWithClient("POST", c.NewURL(gpapi.QueryRoute), c.Client()).
+			EncodeJSON(queryArgs).
+			ParseJSON(res),
+	)
+	err := req.RunWithContext(ctx)
+	if err != nil {
+		return nil, err
```

Comment:
c.f. above: Here you are simply returning the error (as I would have expected)...

Reply:
Sidenote: You can probably skip this `if` statement and just `return res, nil` (since res should be `nil` in case of an error anyway according to Go best practices).